### PR TITLE
[debian] Update debian

### PR DESCRIPTION
**Pull request by stefanhh0**
Closes: https://github.com/muammar/mkchromecast/issues/322

The merge commit contains all commits including the latest commit on master, that is: db2964a1b0d726e03bd3d88833bf5e934a15a3fc.

I have changed some things to the debian specific files so that it is possible to create a debian package that builds mkchromecast without starting immediately in tray mode, since tray mode seems to be broken in debian. Since the old deb Version 0.3.8.1-1 fails with newer pychromecast (see as well: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=965249 and https://github.com/muammar/mkchromecast/issues/322) I gave it a try to use the latest version which includes #325 and found out that it really fixes mkchromecast.

It would be great to get a new deb package uploaded to debian. For now, I have installed a self build package using this PR.

@muammar Please let me know if I can help here any further. Since I am not a debian maintainer / developer I don't know what to do to get the required changes to debian.

### CODE REVIEW

File: debian/compat [hunk 1]
```diff
@@ -1 +0,0 @@
-9
```
Comment (muammar) [inline debian/compat:1]: Why this file was removed?

File: requirements.txt [hunk 2]
```diff
@@ -1,10 +1,8 @@
-PyChromecast==0.8.2
-soco==0.12
-Flask==0.12.2
-netifaces==0.10.6
-setuptools==36.6.0
-requests==2.18.4
-mutagen==1.38
-psutil==5.4.0
+PyGObject
```
Comment (muammar) [inline requirements.txt:1]: I think this dependency creates problems.

File: debian/mkchromecast.links [hunk 3]
```diff
@@ -1 +0,0 @@
-/usr/share/mkchromecast/mkchromecast.py usr/bin/mkchromecast
```
Comment (muammar) [inline debian/mkchromecast.links:1]: Why was this removed?

**Comment by muammar** [comment]
> @muammar Please let me know if I can help here any further. Since I am not a debian maintainer / developer I don't know what to do to get the required changes to debian.

Thanks for this, I really appreciate it. I am a Debian developer (kind of inactive, you know, life), but if you tested this I can upload it.

File: debian/compat (hunk 1 above)
Comment (stefanhh0) [reply debian/compat:1]: It showed warnings that compat 9 is deprecated thus I changed it to: debian/control:  debhelper-compat (=13). Additionally man 7 debhelper states: 

Note that debhelper does not provide debhelper-compat for experimental or beta compatibility levels; packages
experimenting with those compatibility levels should use debian/compat or DH_COMPAT.

So this file is not necessary, however if you like to have it back for experiments I'll re-add it with 13 contained.

File: requirements.txt (hunk 2 above)
Comment (stefanhh0) [reply requirements.txt:1]: This is the result of merging commit: 47ca6a1ad941916965cac456922093a89f0b9c19. 

I have build it using `debuild -b -uc -us -tc` no obvious error was shown. What do you recommend how to change it? I have no clue. I don't know much about this file and python. If the change creates problems, shouldn't it be changed then in the master? Or will it only create problems in the debian branch?

File: debian/mkchromecast.links (hunk 3 above)
Comment (stefanhh0) [reply debian/mkchromecast.links:1]: Well there is no more a `/usr/share/mkchromecast/mkchromecast.py` and because of that a dangling link is created in the resulting deb-file:
```
stefan@mars:~/github$ dpkg-deb -c mkchromecast_0.3.8.1-2_all.deb 
drwxr-xr-x root/root         0 2018-10-21 21:22 ./
drwxr-xr-x root/root         0 2018-10-21 21:22 ./usr/
drwxr-xr-x root/root         0 2018-10-21 21:22 ./usr/bin/
drwxr-xr-x root/root         0 2018-10-21 21:22 ./usr/lib/
[...]
lrwxrwxrwx root/root         0 2018-10-21 21:22 ./usr/bin/mkchromecast -> ../share/mkchromecast/mkchromecast.py
```

I looked into it and found that with commit 3473a77eea7ca989457cbcd13c6c98186dfcf91e mkchromecast was moved directly to bin/mkchromecast, so the file is no longer needed.

I'll add a sentence to my commit.

**Comment by stefanhh0** [comment]
Hello @muammar 

force pushed again, I hope I could answer all your questions. Regarding requirements.txt I am a little unsure, we have to figure out if there is another change required.

I have tested and verified the debian package myself, it is working fine again. So it would be really awesome if you could upload it to debian after this PR has been integrated.

The tray functionality never worked for me, neither in the old package nor in the new package. In the new package the --tray option seems to be ignored completely (In favor of the arg.tray build option maybe?). I guess some commit I merged back to the debian branch changed the behaviour of that command line option. I am not a python developer (and I don't want to dive to deep into python) so I can't fix it. However with this PR at least debian linux gets back casting capabilities integrated into pa. Unfortunately also pulseaudio-dlna is broken and seems to be unmaintained currently. 

Thanks a lot & best regards
Stefan

File: debian/compat (hunk 1 above)
Comment (muammar) [reply debian/compat:1]: If it is in the debian-policy, I am fine with it. Thanks.

File: requirements.txt (hunk 2 above)
Comment (muammar) [reply requirements.txt:1]: It will only create problems when installing the dependencies with pip. But it is not a big deal right now. I would just remove it from the `requirements.txt`.

File: debian/mkchromecast.links (hunk 3 above)
Comment (muammar) [reply debian/mkchromecast.links:1]: Great :)

**Comment by muammar** [comment]
> Hello @muammar
> 
> force pushed again, I hope I could answer all your questions. Regarding requirements.txt I am a little unsure, we have to figure out if there is another change required.
> 

Yes, thanks for doing that. I appreciate it. 

> I have tested and verified the debian package myself, it is working fine again. So it would be really awesome if you could upload it to debian after this PR has been integrated.
> 

I was gonna ask you how you wanted to proceed. Do you want to be part of the uploaders or not?

> The tray functionality never worked for me, neither in the old package nor in the new package. In the new package the --tray option seems to be ignored completely (In favor of the arg.tray build option maybe?). I guess some commit I merged back to the debian branch changed the behaviour of that command line option. I am not a python developer (and I don't want to dive to deep into python) so I can't fix it. However with this PR at least debian linux gets back casting capabilities integrated into pa. Unfortunately also pulseaudio-dlna is broken and seems to be unmaintained currently.
> 

That's odd. What desktop environment are you using? on my workstation that uses awesome WM I am able to use the tray system. Something must have changed either on KDE or Gnome that creates the problem. 

Best,

**Comment by stefanhh0** [comment]
> I was gonna ask you how you wanted to proceed. Do you want to be part of the uploaders or not?

I think it is not necessary  to be part of the uploaders.

> That's odd. What desktop environment are you using? on my workstation that uses awesome WM I am able to use the tray system. Something must have changed either on KDE or Gnome that creates the problem.

I am running KDE. Maybe going to try WM as well when I have some time (currently not so much unfortunately)

File: requirements.txt (hunk 2 above)
Comment (stefanhh0) [reply requirements.txt:1]: Okay that should be done then on master branch and is not relevant for this PR, correct?

**Comment by nm2107** [comment]
Hello guys, thank you for your work here.

Are you sure that the `v0.3.9` release ships with the replacement of the removed `reboot` function ? I still have the errored attempt to use that function :

```bash
$ /usr/bin/mkchromecast -p 10291 --encoder-backend ffmpeg -c wav --sample-rate 44100 --chunk-size 4096

Traceback (most recent call last):
  File "/usr/bin/mkchromecast", line 15, in <module>
    from mkchromecast.cast import casting
  File "/usr/lib/python3.9/site-packages/mkchromecast/cast.py", line 13, in <module>
    from pychromecast.dial import reboot
ImportError: cannot import name 'reboot' from 'pychromecast.dial' (/usr/lib/python3.9/site-packages/pychromecast/dial.py)
```

I'm running fedora 33.
